Brightspace/insights-engagement-dashboard
# Url Bookmark History

## Implementation Notes

The urlState class is a wrapper what observes the state of a filter and parameterizes it for the url. 
This PR adds this class (made by @johngwilkinson ) and hooks it into the chart filters.

## Quality Notes

### Functional Testing
- test cases for all charts and filters will 
  - set a state in the url, verify that the filter initialized with that state.
  - change the state of the filter, verifying that the state is saved as a url parameter.

### Performance
- Probably N/A

### Security
- N/A 

### Dev-Ops
- N/A

### Cost
- N/A

### Documentation
- The pattern is very simple and the wrapper does most of the logic automagically

### UX
- N/A

Does it work in all supported browsers? This doesn't use anything we aren't already using .

Demo: History feature can be demoed at standup.

FYI @anhill-D2L @MykolaGalian @rohitvinnakota @hyehayes @Vitalii-Misechko
